```
../src/nvim/fileio.c:3366:7: error: assigning to 'char *' from 'char_u []' converts between pointers to integer types with different sign [-Werror,-Wpointer-sign]
      SET_ERRMSG_ARG(e_fsync, error);
      ^              ~~~~~~~
../src/nvim/fileio.c:2275:25: note: expanded from macro 'SET_ERRMSG_ARG'
  errnum = NULL, errmsg = msg, errmsgarg = error
```

Reminds me that there was a vim patch to remove all the char_u casting for `emsg`-like functions.

--- a/runtime/doc/eval.txt
+++ b/runtime/doc/eval.txt
@@ -8958,7 +8958,7 @@ winwidth({nr})						*winwidth()*
 		Examples: >
   :echo "The current window has " . winwidth(0) . " columns."
   :if winwidth(0) <= 50
-  :  exe "normal 50\<C-W>|"
+  :  50 wincmd |
   :endif
 <		For getting the terminal or screen size, see the 'columns'
 		option.

--- a/src/nvim/fileio.c
+++ b/src/nvim/fileio.c
@@ -3363,7 +3363,7 @@ restore_backup:
     if (p_fs && (error = os_fsync(fd)) != 0 && !device
         // fsync not supported on this storage.
         && error != UV_ENOTSUP) {
-      SET_ERRMSG_ARG(_("E667: Fsync failed: %s"), error);
+      SET_ERRMSG_ARG(e_fsync, error);
       end = 0;
     }
 

--- a/src/nvim/globals.h
+++ b/src/nvim/globals.h
@@ -958,6 +958,7 @@ EXTERN char_u e_streamkey[] INIT(= N_(
     "E5210: dict key '%s' already set for buffered stream in channel %"
     PRIu64));
 EXTERN char_u e_libcall[] INIT(= N_("E364: Library call failed for \"%s()\""));
+EXTERN char e_fsync[] INIT(= N_("E667: Fsync failed: %s"));
 EXTERN char_u e_mkdir[] INIT(= N_("E739: Cannot create directory %s: %s"));
 EXTERN char_u e_markinval[] INIT(= N_("E19: Mark has invalid line number"));
 EXTERN char_u e_marknotset[] INIT(= N_("E20: Mark not set"));

--- a/src/nvim/testdir/test_writefile.vim
+++ b/src/nvim/testdir/test_writefile.vim
@@ -167,3 +167,10 @@ func Test_writefile_sync_dev_stdout()
     throw 'Skipped: /dev/stdout is not writable'
   endif
 endfunc
+
+func Test_writefile_sync_arg()
+  " This doesn't check if fsync() works, only that the argument is accepted.
+  call writefile(['one'], 'Xtest', 's')
+  call writefile(['two'], 'Xtest', 'S')
+  call delete('Xtest')
+endfunc

